src/group_theory/schur_zassenhaus.lean

```diff
@@ -162,21 +164,204 @@ begin
+    refine h1.coprime_dvd_left _,
+    rw [←nat.mul_dvd_mul_iff_left index_pos, index_mul_card, ←index_map, index_mul_card],
+    exact K.card_quotient_dvd_card },
+  obtain ⟨H, hH⟩ := h2 (quotient_group.quotient K) h5 h6,
+  refine ⟨H.comap (quotient_group.mk' K), _, _⟩,
+  { have key : (N.map (quotient_group.mk' K)).comap (quotient_group.mk' K) = N,
+    { refine comap_map_eq_self _,
+      rwa quotient_group.ker_mk },
+    rwa [←key, comap_sup_eq, hH.symm.sup_eq_top, comap_top] },
+  { rw ← comap_top (quotient_group.mk' K),
+    intro hH',
+    rw [comap_injective this hH', is_complement'_top_right,
+        map_eq_bot_iff, quotient_group.ker_mk] at hH,
+    { exact h4.2 (le_antisymm hK hH) } },
+end
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step3 (K : subgroup N) [(K.map N.subtype).normal] : K = ⊥ ∨ K = ⊤ :=
+begin
+  have key := step2 h1 h2 h3 (K.map N.subtype) K.map_subtype_le,
+  rw ← map_bot N.subtype at key,
+  conv at key { congr, skip, to_rhs, rw [←N.subtype_range, N.subtype.range_eq_map] },
+  have inj := map_injective (show function.injective N.subtype, from subtype.coe_injective),
+  rwa [inj.eq_iff, inj.eq_iff] at key,
+end
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step4 : (fintype.card N).min_fac.prime :=
+(nat.min_fac_prime (N.one_lt_card_iff_ne_bot.mpr (step0 h1 h2 h3)).ne')
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step5 {P : sylow (fintype.card N).min_fac N} : P.1 ≠ ⊥ :=
+begin
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  exact P.ne_bot_of_dvd_card (fintype.card N).min_fac_dvd,
+end
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step6 : is_p_group (fintype.card N).min_fac N :=
+begin
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  refine sylow.nonempty.elim (λ P, P.2.of_surjective P.1.subtype _),
+  rw [←monoid_hom.range_top_iff_surjective, subtype_range],
+  haveI : (P.1.map N.subtype).normal := normalizer_eq_top.mp
+    (step1 h1 h2 h3 (P.1.map N.subtype).normalizer P.normalizer_sup_eq_top),
+  exact (step3 h1 h2 h3 P.1).resolve_left (step5 h1 h2 h3),
+end
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma step7 : is_commutative N :=
+begin
+  haveI := N.bot_or_nontrivial.resolve_left (step0 h1 h2 h3),
+  haveI : fact ((fintype.card N).min_fac.prime) := ⟨step4 h1 h2 h3⟩,
+  exact ⟨⟨λ g h, eq_top_iff.mp ((step3 h1 h2 h3 N.center).resolve_left
+    (step6 h1 h2 h3).bot_lt_center.ne') (mem_top h) g⟩⟩,
+end
+
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+lemma contradiction : false :=
+begin
+  haveI := step7 h1 h2 h3,
+  exact not_exists_of_forall_not h3 (exists_right_complement'_of_coprime_aux h1),
+end
```

Comment:
Please inline this lemma below. Proving `false` looks quite odd, and the variables are far away, so it's not so obvious what's going on at a first glance.

Reply:
What do you mean by inline? Do you mean that I should remove it from the namespace, and add back in the hypotheses?

Reply:
Oh, I think you mean that I should just delete this lemma, and reproduce the proof when needed?

Reply:
Pushed a change.